### PR TITLE
Change struct member names s_addr and s6_addr since it may conflict with predefined macros

### PR DESCRIPTION
**Pull request by sauwming**
Re: #2311

### CODE REVIEW

File: pjlib/include/pj/sock.h [hunk 1]
```diff
@@ -484,15 +484,26 @@ typedef enum pj_socket_sd_type
...
+ */
+//#undef s_addr
+
+/* This macro is used only for backward compatibility in apps which
+ * access pj_in_addr.addr . If it causes any issue, comment this
+ * and access pj_in_addr.addr directly.
+ */
+#ifndef s_addr
+    #define s_addr addr
+#endif
```
Comment (nanangizz) [inline pjlib/include/pj/sock.h]: mention the issue link/number somewhere?

File: pjlib/include/pj/sock.h [hunk 2]
```diff
@@ -535,16 +546,27 @@ struct pj_sockaddr_in
...
+//#undef s6_addr
 
-#undef s6_addr
+/* This macro is used only for backward compatibility in apps which
+ * access pj_in_addr.s6_addr . If it causes any issue, comment this
+ * and access pj_in6_addr.addr directly.
+ */
+#ifndef s6_addr
+    #define s6_addr addr
+#endif
```
Comment (nanangizz) [inline pjlib/include/pj/sock.h]: also this, link.

**Comment by sauwming** [comment]
The important changes are the ones in `pjlib/include/pj/sock.h` and `pjlib/include/pj/compat/socket.h`

Should I change the field name `sin_zero` as well? Unlike `s*addr`, `sin_zero` undefinition can only be found in `compat`, but not in `sock.h`, and it builds fine here on Mac without changing it. Here's the definition of `sin_zero`:
`#define sin_zero __pad /* for BSD UNIX comp`

If it's needed, what should I change it to?

File: pjlib/include/pj/compat/socket.h [hunk 3]
```diff
@@ -61,7 +61,7 @@
  */
 #if defined(_MSC_VER) && defined(PJ_HAS_IPV6) && PJ_HAS_IPV6!=0
 #   ifndef s_addr
-#	define s_addr  S_un.S_addr
+#	define s_addr  S_un.addr
```
Comment (nanangizz) [inline pjlib/include/pj/compat/socket.h]: this should not be changed?

File: pjlib/include/pj/compat/socket.h [hunk 4]
```diff
@@ -179,10 +179,12 @@
 
 /*
  * And undefine these..
+ * Note: unfortunately, this may cause build failure to anyone who
+ * uses these standard macros.
```
Comment (nanangizz) [inline pjlib/include/pj/compat/socket.h]: link to issue

File: pjlib/include/pj/compat/socket.h (hunk 3 above)
Comment (sauwming) [reply pjlib/include/pj/compat/socket.h]: oh no, that's the evil of using batch replace. Need to recheck everything.

**Comment by nanangizz** [comment]
re: renaming field sin_zero, there seems to be no problem on this, so I think it's better to let's just leave it.

**Comment by sauwming** [comment]
Note: This still causes build failure on Windows:
```
`1>Compiling...
1>ssl_sock_ossl.c
1>c:\program files (x86)\microsoft visual studio 8\vc\platformsdk\include\winsock2.h(403) : error C2079: 'sin_addr' uses undefined struct 'in_addr'
1>c:\program files (x86)\microsoft visual studio 8\vc\platformsdk\include\ws2tcpip.h(29) : error C2079: 'imr_multiaddr' uses undefined struct 'in_addr'
```

**Comment by sauwming** [comment]
Closing this PR as we adopt a different approach in #2423.

**Pull request closed without merging**